python/ambassador/envoy/v2/v2listener.py

```diff
@@ -868,13 +868,11 @@ def finalize_http(self) -> None:
             # OK, we have the filter_chain variable set -- build the Envoy virtual_hosts for it.
 
             for host in chain.hosts.values():
-                # This bit of rank paranoia is probably unnecessary: it boils down to 
-                # double-checking host constraints, and making sure that no internal keys
```

Comment:
...and double-checking the host constraints borked the ACME hole-puncher, because we don't synthesize the hole-puncher with any host right now. It's not necessary any more -- after other IHA changes, you can't get a route in a chain anymore without a hostname match.